tests/test_cms_config.py

```diff
@@ -0,0 +1,73 @@
+from unittest.mock import Mock
+
+from django.core.exceptions import ImproperlyConfigured
+
+from cms import app_registration
+from cms.models import Page
+from cms.utils.setup import setup_cms_apps
+
+from djangocms_references.cms_config import ReferencesCMSExtension
+from djangocms_references.test_utils.app_1.models import TestModel1, TestModel2
+from djangocms_references.test_utils.app_2.models import TestModel3, TestModel4
+from djangocms_references.utils import supported_models
+
+from .utils import TestCase
+
+
+class AppRegistrationTestCase(TestCase):
+    def test_missing_cms_config(self):
+        extensions = ReferencesCMSExtension()
+        cms_config = Mock(
+            djangocms_referencess_enabled=True, app_config=Mock(label="blah_cms_config")
+        )
+
+        with self.assertRaises(ImproperlyConfigured):
+            extensions.configure_app(cms_config)
+
+    def test_invalid_cms_config_parameter(self):
+        extensions = ReferencesCMSExtension()
+        cms_config = Mock(
+            djangocms_referencess_enabled=True,
+            reference_models=23234,
+            app_config=Mock(label="blah_cms_config"),
+        )
+
+        with self.assertRaises(ImproperlyConfigured):
+            extensions.configure_app(cms_config)
+
+    def test_valid_cms_config_parameter(self):
+        extensions = ReferencesCMSExtension()
+        cms_config = Mock(
+            djangocms_referencess_enabled=True,
+            reference_models={
+                TestModel1: [],
+                TestModel2: [],
+                TestModel3: [],
+                TestModel4: [],
+            },
+            app_config=Mock(label="blah_cms_config"),
+        )
+
+        with self.assertNotRaises(ImproperlyConfigured):
+            extensions.configure_app(cms_config)
+            register_model = []
+            for model in extensions.references_app_models.keys():
+                register_model.append(model)
+
+            self.assertTrue(TestModel1 in register_model)
+            self.assertTrue(TestModel2 in register_model)
+            self.assertTrue(TestModel3 in register_model)
+            self.assertTrue(TestModel4 in register_model)
```

Comment:
I'm not too sure about creating a var (`register_model`) and then asserting things on that. Always best to keep things simple and as close to what you actually want to know. And I think in this case what you want to know is that when you call `configure_app` on a valid config, the `reference_models` dict is populated as you would expect. And interestingly enough that is never tested :D I mean from these tests I can only deduce the dictionary keys, I have no idea what values you're expecting the dictionary to have.
Also, I'd suggest actually populating at least one or two of the lists in your test data because otherwise how do we know that your code doesn't just always set empty lists for the dict values?